src/Microsoft.ML.TorchSharp/Utils/Range.cs

```diff
@@ -134,5 +137,293 @@ public override string ToString()
             var stop = !range.End.IsFromEnd ? new long?(range.End.Value) : range.End.Value == 0 ? null : new long?(-1 * range.End.Value);
             return TensorIndex.Slice(start, stop);
         }
+
+        /// <summary>
+        /// Slices the specified array using the specified range.
+        /// </summary>
+        public static T[] GetSubArray<T>(T[] array, Range range)
+        {
+            (int offset, int length) = range.GetOffsetAndLength(array.Length);
+
+            if (length == 0)
+            {
+                return Array.Empty<T>();
+            }
+
+            T[] dest = new T[length];
+
+            // Due to array variance, it's possible that the incoming array is
+            // actually of type U[], where U:T; or that an int[] <-> uint[] or
+            // similar cast has occurred. In any case, since it's always legal
+            // to reinterpret U as T in this scenario (but not necessarily the
+            // other way around), we can use Buffer.Memmove here.
+
+            Memmove(
```

Comment:
Instead can you just use https://learn.microsoft.com/en-us/dotnet/api/system.buffer.memorycopy?view=net-7.0?